Change print... to print(...) for new scons release

LindyBalboa opened the pull request:
Scons has switched to using `from __future__ import print_function` which breaks the old print keyword-- it must be a function now. This changes all print statements to functions in the Sconstruct files.

LindyBalboa committed:
Change print... to print(...) for new scons release

sblaisot commented on src/SConscript:
```diff
@@ -618,35 +618,35 @@ def win32_find_program_via_registry(program_name):
+        print("YOU ARE ABOUT TO PACKAGE A DEBUG BUILD!!")
         print
-    print "Binary has size ",
+    print("Binary has size ",)
```
ending this statement with `,)` seems strange. Are you sure this is ok ?

sblaisot commented on src/SConscript:
```diff
@@ -618,35 +618,35 @@ def win32_find_program_via_registry(program_name):
         os.system('ls -lh '+dist_dir+'/mixxx.exe | cut -d \' \' -f 5')
     print
-    print "Installer file ",
+    print("Installer file ",)
```
same here

sblaisot commented on src/SConscript.env:
```diff
@@ -110,8 +110,8 @@ env = conf.Finish()
 
-# Print feature summary
-print "\nFeatures Summary:\n================"
+# print(feature summary)
```
Nope, this is a comment telling the next line "Prints feature summary" ;)

sblaisot commented on src/SConscript.env:
```diff
@@ -166,22 +166,22 @@ if 'msvc' in COMMAND_LINE_TARGETS:
     env.Alias('msvc', msvc)
 
-# Print the build flags (useful if the flags have been cached, ie. if you just run "scons"
+# print(the build flags (useful if the flags have been cached, ie. if you just run "scons")
```
Nope, english comment here also, nos commented code.

LindyBalboa replied:
Ahhh thanks for the review. This is where sed bites you. I submitted the PR after getting home late last night just to do it. I will clean it up.

It /should/ remain compatible with older versions of scons. Python 2.7 handles the print() function fine, but with the import from __future__ it complain if the keyword version is used.

sblaisot reviewed (changes requested):
Thanks for working on this.

Do you know if your changes are compatible with older versions of scons or will it break thinks for people that have older scons or python ?

LindyBalboa commented:
Ahhh thanks for the review. This is where sed bites you. I submitted the PR after getting home late last night just to do it. I will clean it up.

It /should/ remain compatible with older versions of scons. Python 2.7 handles the print() function fine, but with the import from future it complain if the keyword version is used.

sblaisot commented on src/SConscript:
```diff
@@ -618,35 +618,35 @@ def win32_find_program_via_registry(program_name):
+        print("YOU ARE ABOUT TO PACKAGE A DEBUG BUILD!!")
         print
-    print "Binary has size ",
+    print("Binary has size ")
```
extra carriage return here.
Expected result on screen is
```
Binary has size 31888384
```

Actual result is 
```
Binary has size 
31888384
```

see line 11485 to 11492 of https://ci.appveyor.com/project/mixxxdj/mixxx/build/master-2608/job/buaxyoi627i7uud8

Use this form instead:
```python
print("Binary has size ", end='')
```

sblaisot replied:
for this one, you can't automate it when comma at the end of line because of https://github.com/mixxxdj/mixxx/blob/master/src/SConscript#L1021

sblaisot commented on src/SConscript:
```diff
@@ -618,35 +618,35 @@ def win32_find_program_via_registry(program_name):
         os.system('ls -lh '+dist_dir+'/mixxx.exe | cut -d \' \' -f 5')
     print
-    print "Installer file ",
+    print("Installer file ")
```
same here

sblaisot commented on src/SConscript:
```diff
@@ -657,16 +657,16 @@ def BuildRelease(target, source, env):
         if build.toolchain_is_msvs:
             redist_file = 'vc_redist.%s.exe' % arch
-            print "Searching for the Visual C++ DLL installer package", redist_file
+            print("Searching for the Visual C++ DLL installer package", redist_file)
```
`print("Searching for the Visual C++ DLL installer package " + redist_file)`

sblaisot commented on src/SConscript:
```diff
@@ -955,30 +955,30 @@ def ubuntu_cleanup():
+        print("==== Mixxx Post-Build Checks ====")
         print
-        print "You have built version ", mixxx_version
+        print("You have built version ", mixxx_version)
```
`,` -> `+`

sblaisot commented on src/SConscript:
```diff
@@ -955,30 +955,30 @@ def ubuntu_cleanup():
 
-        print "Now building DEB package..."
+        print("Now building DEB package...")
         print
```
print()

sblaisot commented on src/SConscript:
```diff
@@ -1011,21 +1011,21 @@ def BuildUbuntuPackage(target, source, env):
-        print
+        print("* Exporting source folder from current workspace (%s rev: %s)" % (vcs_name,
+                                                                                 vcs_revision))
+        print("\n")
```
`print()` should be sufficient.
`print('\n')` adds an extra carriage return

sblaisot reviewed (changes requested):
here is a couple of comments to address.

You probably also need to update all python files in the `build` subdirectory.

daschuer commented:
compiling on Linux fails with 
```
Checking for C library vamp-sdk... yes
File "/home/travis/build/mixxxdj/mixxx/src/SConscript", line 624
print("Binary has size ", end='')
^
SyntaxError: invalid syntax
```

I think
```phyton 
from __future__ import print_function
```
will fix the issue. 
This seems to be available from Python 2.6. Travis is on Trusty with Python 2.7 so it should work.

LindyBalboa committed:
Add from __future__ import print_function to Sconscript.

Currently the file only gets the environmental python, which is
typically python2 but sometimes python3. Perhaps the shebang line should
specify python2? Either way, the import works in python3 as well so it
should not be an issue for now.

LindyBalboa commented:
Ah even though I was calling scons with the python2 executable, it must somehow call the sconscript file in a subprocess which, given the shebang, uses the environmental python. On Arch Linux this happens to be python3 so that is why I didn't get an error while building. Added the import and apparently it doesn't cause python3 to complain so it should be fine.

daschuer commented:
Nice, the build servers are happy, so we can give this a go.
Thank you for your work!
